src/components/PortalModalBtn.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { BaseSyntheticEvent, FC, ReactElement } from "react";
+import usePortal from "react-useportal";
+
+interface Props {
+  renderModal: (closeModal: () => void) => ReactElement;
+  renderButton: (
+    openModal: (e: BaseSyntheticEvent<object, unknown, unknown>) => void,
+  ) => ReactElement;
+}
+
+const PortalModalBtn: FC<Props> = ({ renderButton, renderModal }) => {
```

Comment:
I see this is to reduce boilerplate code. I think it is trading readability for it. Because this component is coupling the renderModal and renderButton with its own local functions by passing them back to the caller, that is passing them on too. So the flow of dependencies is going in, then going out and then in again. I hope you know what I mean? Maybe better to keep the usePortal call in each specific button?